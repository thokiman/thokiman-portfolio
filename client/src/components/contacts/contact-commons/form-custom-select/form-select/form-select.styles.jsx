import styled from 'styled-components';

const maxWidthSmartphone = 800;

export const FormGroupSelectContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 98%;
  margin: 15px 10px 40px;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    margin: 30px 10px 40px;
  }
`;

FormGroupSelectContainer.displayName = 'FormGroupSelectContainer';
