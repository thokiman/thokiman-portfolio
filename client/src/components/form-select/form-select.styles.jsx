import styled from "styled-components";

export const FormGroupSelectContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 98%;
  margin: 15px 10px 40px;
`;

FormGroupSelectContainer.displayName = "FormGroupSelectContainer";
