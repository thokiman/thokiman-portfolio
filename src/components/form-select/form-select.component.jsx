import React from "react";
import { HiArrowCircleLeft } from "react-icons/hi";
import { FaCheckCircle, FaTimesCircle } from "react-icons/fa";

import "./form-select.styles.scss";

const FormInputSelect = ({ handleType, isTypeClick, handleClick, value }) => {
  return (
    <div className="group-select-container">
      <div className={`select-title-container${value ? " active" : ""}`}>
        {value ? (
          <div className="select-title-active">Type of Inquiry :</div>
        ) : (
          ""
        )}
      </div>
      <div onClick={handleType} className="select-value">
        {
          <span className="checker-area-code">
            {value ? (
              <FaCheckCircle className="checker-icon" />
            ) : (
              <FaTimesCircle className="checker-icon" />
            )}
          </span>
        }
        {value ? (
          <div className="select-value-active">{value}</div>
        ) : (
          <div className="select-title">Type of Inquiry :</div>
        )}
        <span className={`select-arrow-button${isTypeClick ? " active" : ""}`}>
          <HiArrowCircleLeft />
        </span>
      </div>
      {isTypeClick ? (
        <ul className="select-container">
          <li className="select-item" onClick={handleClick}>
            Engineering
          </li>
          <li className="select-item" onClick={handleClick}>
            Web/Mobile Development
          </li>
          <li className="select-item" onClick={handleClick}>
            Graphic Design
          </li>
          <li className="select-item" onClick={handleClick}>
            Photography
          </li>
        </ul>
      ) : (
        ""
      )}
    </div>
  );
};

export default FormInputSelect;
