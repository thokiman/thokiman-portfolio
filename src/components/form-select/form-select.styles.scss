.group-select-container {

    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 98%;
    margin: 15px 10px 40px;


    .select-title-active {

        text-align: start;
        width: 100%;
        font-size: 30px;
        color: #bccfd5;
        margin-bottom: 5px;
        animation: slide-in-fwd-bl 0.4s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;

        @keyframes slide-in-fwd-bl {
            0% {
                transform: translateZ(-20px) translateY(20px) translateX(-20px);
                opacity: 0;
            }

            100% {
                transform: translateZ(0) translateY(0) translateX(0);
                opacity: 1;
            }
        }
    }

    .select-title-container {
        display: block;
        height: 0px;
        transition: height 0.5s ease-in-out;

        &.active {
            display: block;
            height: 39px;
            transition: height 0.5s ease-in-out;
        }

    }

    .select-value {
        position: relative;
        right: 10px;
        width: 100%;
        display: flex;
        flex-direction: row;
        font-size: 35px;
        color: #bccfd5;
        cursor: pointer;


        &:hover {

            .select-title,
            .select-arrow-button {
                color: #bccfd5;
                transition: color linear 0.2s;
            }
        }

        .checker-area-code,
        .checker-phone-number {
            position: relative;
            top: 10px;
            right: 7px;
            color: #128194;

            .checker-icon {
                width: 20px;
                height: 20px;
            }
        }

        .select-title {
            font-size: 35px;
            text-align: start;
            width: 90%;
            color: #128194;
            position: relative;
            right: 5px;
            padding: 5px;
            border-bottom: #128194 solid 1px;
        }

        .select-value-active {
            font-size: 35px;
            text-align: center;
            width: 90%;
            color: #128194;
            padding: 5px;
            animation: slide-in-fwd-bl 0.4s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
            border-bottom: #128194 solid 1px;

            @keyframes slide-in-fwd-bl {
                0% {
                    transform: translateZ(-20px) translateY(20px) translateX(-20px);
                    opacity: 0;
                }

                100% {
                    transform: translateZ(0) translateY(0) translateX(0);
                    opacity: 1;
                }
            }

        }

        .select-arrow-button {

            position: relative;
            left: 10px;
            display: flex;
            align-items: center;
            width: 35px;
            color: #128194;
            transition: all ease-in-out 0.4s;

            &.active {
                animation: arrow-icon ease-in-out 0.4s;
                transform: rotate(-90deg);

                @keyframes arrow-icon {
                    from {
                        transform: rotate(0deg)
                    }

                    to {
                        transform: rotate(-90deg);
                    }
                }
            }
        }
    }

    .select-container {

        list-style-type: none;
        margin: 5px 0;
        padding: 0;
        width: 100%;

        .select-item {

            cursor: pointer;
            width: 100%;
            animation: swing-in-top-fwd 1s cubic-bezier(0.175, 0.885, 0.320, 1.275) both;
            font-size: 25px;
            color: #bccfd5;
            margin-bottom: 5px;

            &:hover {
                color: #128194;
                transition: color linear 0.2s;
            }

            @keyframes swing-in-top-fwd {
                0% {
                    transform: rotateX(-100deg);
                    transform-origin: top;
                    opacity: 0;

                }

                100% {
                    transform: rotateX(0deg);
                    transform-origin: top;
                    opacity: 1;

                }
            }
        }
    }
}