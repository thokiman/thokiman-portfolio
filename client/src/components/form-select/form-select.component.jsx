import React from "react";
import FormInputSelectElement from "../form-select-element/form-select-element.component";
import FormInputSelectLabel from "../form-select-label/form-select-label.component";
import FormInputSelectOptions from "../form-select-options-element/form-select-options-element.component";

import { FormGroupSelectContainer } from "./form-select.styles";

export const FormInputSelect = ({
  handleType,
  isTypeClick,
  handleClick,
  value,
}) => {
  return (
    <FormGroupSelectContainer>
      <FormInputSelectLabel value={value} />
      <FormInputSelectElement
        handleType={handleType}
        isTypeClick={isTypeClick}
        value={value}
      />
      {isTypeClick ? <FormInputSelectOptions handleClick={handleClick} /> : ""}
    </FormGroupSelectContainer>
  );
};

export default FormInputSelect;
