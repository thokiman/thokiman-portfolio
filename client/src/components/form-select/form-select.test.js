import React from "react";
import { shallow } from "enzyme";

import { FormInputSelect } from "./form-select.component";

describe("FormInputSelect component", () => {
  let wrapper;
  let mockHandleType;
  let mockHandleClick;
  const mockFalseBool = false;

  beforeEach(() => {
    mockHandleType = jest.fn();
    mockHandleClick = jest.fn();
    const mockProps = {
      handleType: mockHandleType,
      isTypeClick: mockFalseBool,
      handleClick: mockHandleClick,
      value: "Engineering",
    };
    wrapper = shallow(<FormInputSelect {...mockProps} />);
  });

  it("should render FormInputSelect component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
