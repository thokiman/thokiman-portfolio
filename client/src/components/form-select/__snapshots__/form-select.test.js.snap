// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputSelect component should render FormInputSelect component 1`] = `ShallowWrapper {}`;
